pkg/controller/clusters/status.go

```diff
@@ -215,6 +222,13 @@ func updateWorkersReadyCondition(cluster *anywherev1.Cluster, machineDeployments
 			return
 		}
 
+		// Skip updating the replicas for the machine deployments which have autoscaling configuration annotation
+		if md.ObjectMeta.Annotations != nil {
+			if _, ok := md.ObjectMeta.Annotations[clusterapi.NodeGroupMinSizeAnnotation]; ok {
```

Comment:
Should we check for the existence of either min or max? We shouldn't assume that both need to be specified in case we change the requirement in the future, unless autoscaler requires both in general on the MD object

Reply:
If both min and max are not specified, our controller considers it as autoscaling configuration not being set and does not add autoscaler annotations to the md object. So we can assume that atleast one of them must be specified. By default, capi defaults the min and max count to 0 if not specified. So if only min is specified, then max will be defaulted to 0 and min must be set to 0 as we have a validation which checks that min is less than or equal to max. In either case where only of them is specified, there will be annotations for both of them on the md object so it should be fine to check for existence on either of min or max.

Reply:
It will be fine for us, but let's say we change that behavior in the future. Is it a requirement for autoscaler for both to exist on the MD object?

Reply:
Yes, autoscaler validates that both annotations exist on the md object for that worker node group to be considered a scalable resource. 
Ref:- https://github.com/kubernetes/autoscaler/blob/83db225ce2c4b368a8ee30152691746ed9e6d5ae/cluster-autoscaler/cloudprovider/clusterapi/clusterapi_utils.go#L119C1-L149C2